### DISCUSSION TITLE
Why use Stash Bucket

### DISCUSSION THREAD

**Discussion by JcKendo** [dragonflydb/dragonfly#4305] [Q&A]
Hi Team
I viewed the source code and read about the stash bucket. I don't understand why it is needed; I understand that split is active when all buckets and stash buckets are full.  Because I think stash buckets are not necessary, Stash Buckets are as swam memory.
If I misunderstood, please help clarify and explain in detail.

**Comment by romange** [comment]
Hi there,

Please read the Dash paper: https://arxiv.org/pdf/2003.07302

the short answer is that stash records are commonly used for hash tables to fight collisions  and improve table utilisation.

**Comment by JcKendo** [comment]
Thanks,
